dask_labextension/dashboardhandler.py

```diff
@@ -73,6 +81,13 @@ async def get(self, url) -> None:
 
             individual_plots = json.loads(individual_plots_response.body)
 
+            # If there was query in original URL, append to URLs returned
+            if query:
+                for name, plot_url in individual_plots.items():
+                    individual_plots[name] = f"{plot_url}{query}"
+                url = f"{url}{query}"
+                effective_url = f"{url}{query}"
```

Comment:
I don't think this logic for `effective_url` is correct:

1. `effective_url` could be different from `url` if there is a redirect in the initial `url` (some deployments have had that in the past, and we needed to follow the redirect to get the right individual plot urls).
2. `url` is updated on the previous line, so this *always* appends the query twice to `effective_url`, which you can confirm by inspecting the network tab.

So I think the correct logic would be something like:

```python
    if effective_url:
        effective_url = f"{effective_url}{query}"
```

Reply:
Yep, thanks! That was entirely a copy-paste error.